l10n_it_invoices_data_communication/models/communication.py

```diff
@@ -0,0 +1,2538 @@
+                  ('fiscal_document_type_id.out_refund', '=', True),
+                  ]
+        return domain
+
+    def _get_fatture_emesse(self):
+        self.ensure_one()
+        domain = self._get_fatture_emesse_domain()
+        return self.env['account.invoice'].search(domain)
+
+    def _prepare_cedenti_dati_fatture(self, fatture_ricevute, cedenti):
+        dati_fatture = []
+        posizione = 0
+        for cedente in cedenti:
+            # Fatture
+            fatture = fatture_ricevute.filtered(
+                lambda fr: fr.partner_id.id == cedente.id)
+            vals_fatture = []
+            for fattura in fatture:
+                posizione += 1
+                val = {
+                    'posizione': posizione,
+                    'invoice_id': fattura.id,
+                    'dati_fattura_TipoDocumento':
+                        fattura.fiscal_document_type_id.id,
+                    'dati_fattura_Data': fattura.date_invoice,
+                    'dati_fattura_DataRegistrazione':
+                        fattura.date_invoice,
+                    'dati_fattura_Numero': self._parse_fattura_numero(
+                        fattura.reference) or '',
+                    'dati_fattura_iva_ids':
+                        fattura._get_tax_comunicazione_dati_iva()
+                }
+                val = self._prepare_fattura_ricevuta(val, fattura)
+                vals_fatture.append((0, 0, val))
+
+            val_cedente = {
+                'partner_id': cedente.id,
+                'fatture_ricevute_body_ids': vals_fatture
+            }
+            vals = self._prepare_cedente_partner_id(
+                cedente)
+            val_cedente.update(vals)
+            dati_fatture.append((0, 0, val_cedente))
+        return dati_fatture
+
+    def compute_fatture_ricevute(self):
+        self.ensure_one()
+        fatture_ricevute = self._get_fatture_ricevute()
+        if fatture_ricevute:
+            self.partner_cessionario_id = \
+                fatture_ricevute[0].company_id.partner_id.id
+            self.onchange_partner_cessionario_id()
+
+            cedenti = fatture_ricevute.mapped('partner_id')
+            dati_fatture = self._prepare_cedenti_dati_fatture(
+                fatture_ricevute, cedenti)
+            self.fatture_ricevute_ids = dati_fatture
+
+    def _get_fatture_ricevute_domain(self):
+        domain = [('comunicazione_dati_iva_escludi', '=', True)]
+        no_journal_ids = self.env['account.journal'].search(domain).ids
+        domain = [('type', 'in', ['in_invoice', 'in_refund']),
+                  ('comunicazione_dati_iva_escludi', '=', False),
+                  ('move_id', '!=', False),
+                  ('move_id.journal_id', 'not in', no_journal_ids),
+                  ('company_id', '=', self.company_id.id),
+                  ('date_invoice', '>=', self.date_start),
+                  ('date_invoice', '<=', self.date_end),
```

Comment:
se filtro le fatture ricevute per `date_invoice` saranno escluse quelle ricevute nel mese di mesi precedenti, il campo da usare sarebbe `registration_date` (che diventa `date` nella 10.0)

Reply:
@sergiocorato non ho capito bene, il campo cui ti riferisci è
https://github.com/OCA/l10n-italy/blob/f331c7242fb54d269b2eefc153f375214ad2e598/account_invoice_entry_date/models/account.py#L33
giusto?

Quindi mi stai dicendo che:

- Nella 8.0 per filtrare le fatture dovrei usare il valore del campo `registration_date`, aggiunto da un modulo OCA. 
  **E se fosse vuoto perché il modulo non era installato quando è stata validata la fattura?**
- Lo scopo di `registration_date` è stato sostituito dal campo `date` da Odoo core, quindi il campo `registration_date` è stato migrato o altro, comunque non è più utilizzato in v10
- Nella 10.0 non va bene comunque usare il campo `date_invoice` ma dovremmo invece usare il campo `date`

Reply:
Sì nella 10 il campo è `date`, direi che nella 8 se non ci fosse il campo `registration_date` sia da usare il `date_invoice`, anche se è abbastanza poco plausibile che venga fatto un invio di dati fatture con registrazioni fatture di acquisto fatte sulla data fattura
Di per sè il campo sarebbe pari a `move_id.date`, che chiaramente però potrebbe essere modificato dall'utente
Forse è più teoria che pratica ;)

Reply:
Uso `registration_date` se c'è, altrimenti `date_invoice`